boa/src/builtins/mod.rs

```diff
@@ -22,30 +22,42 @@ pub(crate) use self::{
     bigint::BigInt,
     boolean::Boolean,
     error::{Error, RangeError, TypeError},
-    function::Function,
+    json::Json,
+    math::Math,
+    nan::NaN,
     number::Number,
     regexp::RegExp,
     string::String,
+    symbol::Symbol,
     value::{ResultValue, Value},
 };
 
 /// Initializes builtin objects and functions
 #[inline]
 pub fn init(global: &Value) {
-    Array::init(global);
-    BigInt::init(global);
-    Boolean::init(global);
-    json::init(global);
-    math::init(global);
-    nan::init(global);
-    Number::init(global);
-    object::init(global);
-    function::init(global);
-    RegExp::init(global);
-    String::init(global);
-    symbol::init(global);
-    console::init(global);
-    Error::init(global);
-    RangeError::init(global);
-    TypeError::init(global);
+    let globals = vec![
+        // The `Function` global must be initialized before other types.
+        function::init(global),
+        Array::init(global),
+        BigInt::init(global),
+        Boolean::init(global),
+        Json::init(global),
+        Math::init(global),
+        NaN::init(global),
+        Number::init(global),
+        object::init(global),
+        RegExp::init(global),
+        String::init(global),
+        Symbol::init(global),
+        console::init(global),
+        // Global error types.
+        Error::init(global),
+        RangeError::init(global),
+        TypeError::init(global),
+    ];
+
+    let mut global_object = global.as_object_mut().expect("global object");
```

Comment:
I'm guessing that the `global` object is always an object, right? So why don't we just pass the "object" around instead of the `Value`?

Reply:
Hmmm. That's a good point! I will see what I can do :)